api/v1alpha1/ratelimit_types.go

```diff
@@ -42,17 +54,45 @@ type GlobalRateLimit struct {
 	Rules []RateLimitRule `json:"rules"`
 }
 
+// LocalRateLimit defines local rate limit configuration.
+type LocalRateLimit struct {
+
+	// Envoy requires a default rate limit to be set for each route.
+	// The other possible option is to set the default rate limit to unit32 max and set the
+	// fill interval to a short period, like 1 second. This will effectively make
+	// the default rate limit "unlimited".
+	// See https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#using-rate-limit-descriptors-for-local-rate-limiting
+
+	// Limit is the default rate limit for a route if no rules match.
+	// If a request does not match any of the rules, it is counted towards this limit.
+	//
+	// Note: Limit is applied per route. Even if a policy targets a gateway,
+	// each route in that gateway still has a separate rate limit bucket.
+	// For example, if a gateway has 2 routes, and the limit is 100r/s, then
+	// each route has its own 100r/s rate limit bucket.
+	Limit RateLimitValue `json:"default"`
+
+	// Rules are a list of RateLimit selectors and limits. They're used to define
+	// fine-grained rate limits that can be applied to specific clients using
+	// attributes from the traffic flow.
+	//
+	// Orders matters here as the rules are processed sequentially.
+	// The first rule that matches the request is applied.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	Rules []RateLimitRule `json:"rules"`
+}
+
 // RateLimitRule defines the semantics for matching attributes
 // from the incoming requests, and setting limits for them.
 type RateLimitRule struct {
 	// ClientSelectors holds the list of select conditions to select
 	// specific clients using attributes from the traffic flow.
 	// All individual select conditions must hold True for this rule
 	// and its limit to be applied.
-	// If this field is empty, it is equivalent to True, and
-	// the limit is applied.
 	//
-	// +optional
+	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=8
 	ClientSelectors []RateLimitSelectCondition `json:"clientSelectors,omitempty"`
```

Comment:
I propose renaming `ClientSelectors` to `Matches` and `RateLimitSelectCondition` to `RateLimitMatch`

The rationale behind this suggestion is to align with the Gateway API's terminology and make it more intuitive for users. `Selector` is Envoy-specific, and using consistent terminology like `Matches` in both `RateLimit` and `HTTPRoute` enhances clarity for users interacting with the API.

Reply:
this was a long discussion, the original API I proposed used `matches` for the exact reason you are suggesting but others wanted to highlight the distinction, more in https://github.com/envoyproxy/gateway/pull/706

Reply:
I may missed it but I didn't find the discussion around `matches` vs `selectors`. Could you please point me to the comments? Thanks!

I'm not sure how it sounds to native speakers, but from a non-native speaker's perspective, I was really bothered by the term `Selector` when I first looked at the rate limit API.